### DISCUSSION TITLE
A basic working wizard example with types on 4.x.x

### DISCUSSION THREAD

**Discussion by FarazPatankar** [telegraf/telegraf#1352] [Q&A]
Hey, so I've spent a lot of time trying to get something to work without TypeScript shouting at me but have had no success so far. Could someone help me figure this out?

For simplicity, let's take a basic example. Let's say I want to create a bot that allows you to convert currencies and also view live currency rates.

For this, I'd start with:
```js
import express from 'express'
import { Markup, Telegraf } from 'telegraf'

const app = express();
const PORT = 8000;

const bot = new Telegraf(TOKEN_GOES_HERE)
bot.start(ctx => ctx.reply('Hey!'))

bot.command('money', ctx => {
  const options = Markup.inlineKeyboard(
    [
      Markup.button.callback('Currency converter', 'CURRENCY_CONVERTER'),
      Markup.button.callback('Live rates', 'LIVE_RATES'),
    ],
  )
  ctx.reply('Money command options:', options)
})

bot.launch()

app.listen(PORT, () => {
  console.log(`⚡️[server]: Server is running at https://localhost:${PORT}`);
});
```

All good till here but this where I get completely lost. I've tried about 20 different things but can't get things to run without using `any`.

When I write the command to enter my scene, I see an error:
```js
bot.action('CURRENCY_CONVERTER', ctx => ctx.scene.enter('CURRENCY_CONVERTER'))

// Property 'scene' does not exist on type 'GuardedContext<Context & { match: RegExpExecArray; }, Record<"callback_query", object>>'
```

I am also seeing similar issues whenever I try to create a simple wizard scene. All I want to do is ask the user three things:
1. What currency do they want to convert from?
2. What currency do they want to convert to?
3. What is the amount they'd like to convert?

After this, I'll call an API and just send them the response. It'd also be great if someone could tell me how to handle errors or give options to quit mid-scene.

Also, does anything change if I have multiple commands and multiple scenes associated with each command?

Any help would be greatly appreciated. Thanks!

**Comment by sachdeva-shrey** [comment]
Hey @wojpawlik, could you please share an example here? We are also facing the same issue, implementing wizards and scenes.
